src/Common/FileCache.cpp

```diff
@@ -191,6 +194,34 @@ FileSegments LRUFileCache::getImpl(
     return result;
 }
 
+FileSegments LRUFileCache::splitRangeIntoEmptyCells(
+    const Key & key, size_t offset, size_t size, std::lock_guard<std::mutex> & cache_lock)
```

Comment:
This tiny optimization from commit [6b60f7c](https://github.com/ClickHouse/ClickHouse/pull/33717/commits/6b60f7c57fabf370536ea962e7adb636531da54c) seems to improve first queries quite well
(both columns are first queries (without any initial cache) and both do download into cache on first access)

![Screenshot 2022-02-20 at 13 51 01](https://user-images.githubusercontent.com/54203879/154843350-9a77e278-f736-46a3-b8b0-4e61a785f58c.png)